server/src/models/mod.rs

```diff
@@ -12,134 +12,268 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+//! Implementation of the model types used for compressing time series segments
+//! as models and functions for efficiently computing aggregates from models of
+//! each type. The module itself contains general functionality used by the
+//! model types.
+
 mod gorilla;
 mod pmcmean;
 mod swing;
 
-use datafusion::arrow::array::{
-    Float32Builder, Int32Array, Int32Builder, Int64Array, TimestampMillisecondBuilder,
+use std::cmp::{Ordering, PartialOrd};
+
+use datafusion::arrow::array::{Int32Array, Int64Array};
+
+use crate::errors::MiniModelarDBError;
+use crate::types::{
+    TimeSeriesId, TimeSeriesIdArray, TimeSeriesIdBuilder, Timestamp, TimestampBuilder, Value,
+    ValueBuilder,
 };
 
-// TODO: can mtid be converted to a model type enum without adding overhead?
+/// Unique ids for each model type. Constant values are used instead of an enum
+/// so the stored model type ids can be used in match expressions without being
+/// converted to an enum first. Zero and one is not used for compatibility with
```

Comment:
Very minor but maybe "0 and 1 are.." instead.

Reply:
Fixed.